data/tutorials/lg_02_if_statements_loops_and_recursion.md

```diff
@@ -1019,13 +1026,17 @@ val range2 : int -> int -> int list -> int list = <fun>
 val range : int -> int -> int list = <fun>
 ```
 That was a brief overview of tail recursion, but in real world
-situations determining if a function is tail recursive can be quite
-hard. What did we really learn here? One thing is that recursive
+situations, determining if a function is tail-recursive can be quite
+hard. 
+
+What did we really learn here? 
+
+One thing is that recursive
 functions have a dangerous trap for inexperienced programmers. Your
-function can appear to work for small inputs (during testing), but fail
+function can appear to work for small inputs (during testing), but then fail
 catastrophically in the field when exposed to large inputs. This is one
-argument *against* using recursive functions, and for using explicit
-while loops when possible.
+argument *against* using recursive functions, but for using explicit
+`while` loops when possible.
```

Comment:
Maybe adding "instead" in the 2nd part of the sentence would be clearer?

Reply:
Like: "instead, use explicit `while` loops when possible" ?

Reply:
Yes for example.

Or "... and for using explicit `while` loops when possible instead" ?